src/c-writer.cc

```diff
@@ -2727,12 +2728,10 @@ void CWriter::Write(const ExprList& exprs) {
           Write(", ", FuncTypeExpr(func_type), ", ", StackVar(0));
           Write(", ", ExternalInstanceRef(ModuleFieldType::Table, table->name),
                 ".data[", StackVar(0), "].module_instance");
-          if (num_params > 0) {
-            Write(", ");
-          }
+          needs_initial_comma = true;
         }
         for (Index i = 0; i < num_params; ++i) {
-          if (i > 0) {
+          if (needs_initial_comma || i > 0) {
             Write(", ");
           }
```

Comment:
you can also do `else { needs_comma = true; }` and drop `i > 0` above.

Reply:
done